pkg/component/logging/vali/vali.go

```diff
@@ -0,0 +1,1104 @@
+					StabilizationDuration: pointer.String("5m"),
+					MinChange: hvpav1alpha1.ScaleParams{
+						CPU: hvpav1alpha1.ChangeParams{
+							Value:      pointer.String("100m"),
+							Percentage: pointer.Int32(80),
+						},
+						Memory: hvpav1alpha1.ChangeParams{
+							Value:      pointer.String("300M"),
+							Percentage: pointer.Int32(80),
+						},
+					},
+				},
+				ScaleDown: hvpav1alpha1.ScaleType{
+					UpdatePolicy: hvpav1alpha1.UpdatePolicy{
+						UpdateMode: pointer.String(hvpav1alpha1.UpdateModeAuto),
+					},
+					StabilizationDuration: pointer.String("168h"),
+					MinChange: hvpav1alpha1.ScaleParams{
+						CPU: hvpav1alpha1.ChangeParams{
+							Value:      pointer.String("200m"),
+							Percentage: pointer.Int32(80),
+						},
+						Memory: hvpav1alpha1.ChangeParams{
+							Value:      pointer.String("500M"),
+							Percentage: pointer.Int32(80),
+						},
+					},
+				},
+				LimitsRequestsGapScaleParams: hvpav1alpha1.ScaleParams{
+					CPU: hvpav1alpha1.ChangeParams{
+						Value:      pointer.String("300m"),
+						Percentage: pointer.Int32(40),
+					},
+					Memory: hvpav1alpha1.ChangeParams{
+						Value:      pointer.String("1G"),
+						Percentage: pointer.Int32(40),
+					},
+				},
+				Template: hvpav1alpha1.VpaTemplate{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							v1beta1constants.LabelRole: valiName + "vpa",
+						},
+					},
+					Spec: hvpav1alpha1.VpaTemplateSpec{
+						ResourcePolicy: &vpaautoscalingv1.PodResourcePolicy{
+							ContainerPolicies: []vpaautoscalingv1.ContainerResourcePolicy{
+								{
+									ContainerName: valiName,
+									MinAllowed: corev1.ResourceList{
+										corev1.ResourceMemory: resource.MustParse("300M"),
+									},
+									MaxAllowed: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse("800m"),
+										corev1.ResourceMemory: resource.MustParse("3Gi"),
+									},
+									ControlledValues: &controlledValues,
+								},
+								{
+									ContainerName:    curatorName,
+									Mode:             &containerPolicyOff,
+									ControlledValues: &controlledValues,
+								},
+								{
+									ContainerName:    initLargeDirName,
+									Mode:             &containerPolicyOff,
+									ControlledValues: &controlledValues,
+								},
+							},
+						},
+					},
+				},
+			},
+			WeightBasedScalingIntervals: []hvpav1alpha1.WeightBasedScalingInterval{
+				{
+					VpaWeight:         hvpav1alpha1.VpaOnly,
+					StartReplicaCount: v.values.Replicas,
+					LastReplicaCount:  v.values.Replicas,
+				},
+			},
+			TargetRef: &autoscalingv2beta1.CrossVersionObjectReference{
+				APIVersion: "apps/v1",
+				Kind:       "StatefulSet",
+				Name:       valiName,
+			},
+		},
+	}
+
+	if v.values.MaintenanceBegin != "" && v.values.MaintenanceEnd != "" {
+		obj.Spec.MaintenanceTimeWindow = &hvpav1alpha1.MaintenanceTimeWindow{
+			Begin: v.values.MaintenanceBegin,
+			End:   v.values.MaintenanceEnd,
+		}
```

Comment:
Why not passing a `&hvpav1alpha1.MaintenanceTimeWindow` to the values instead of the begin and end only?

Reply:
I was not sure which will be more acceptable primitives or struct